test/fixtures/gcs-fixture/src/main/java/fixture/gcs/GoogleCloudStorageHttpHandler.java

```diff
@@ -266,56 +271,62 @@ private String httpServerUrl(final HttpExchange exchange) {
         return "http://" + InetAddresses.toUriString(address.getAddress()) + ":" + address.getPort();
     }
 
+    private static final Pattern NAME_PATTERN = Pattern.compile("\"name\":\"([^\"]*)\"");
+
     public static Optional<Tuple<String, BytesArray>> parseMultipartRequestBody(final InputStream requestBody) throws IOException {
         Tuple<String, BytesArray> content = null;
         try (BufferedInputStream in = new BufferedInputStream(new GZIPInputStream(requestBody))) {
             String name = null;
             int read;
+            ByteArrayOutputStream out = new ByteArrayOutputStream() {
+                @Override
+                public byte[] toByteArray() {
+                    return buf;
+                }
+            };
             while ((read = in.read()) != -1) {
+                out.reset();
                 boolean markAndContinue = false;
-                try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
-                    do { // search next consecutive {carriage return, new line} chars and stop
-                        if ((char) read == '\r') {
-                            int next = in.read();
-                            if (next != -1) {
-                                if (next == '\n') {
-                                    break;
-                                }
-                                out.write(read);
-                                out.write(next);
-                                continue;
+                do { // search next consecutive {carriage return, new line} chars and stop
+                    if ((char) read == '\r') {
+                        int next = in.read();
+                        if (next != -1) {
+                            if (next == '\n') {
+                                break;
                             }
-                        }
-                        out.write(read);
-                    } while ((read = in.read()) != -1);
-
-                    final String line = new String(out.toByteArray(), UTF_8);
```

Comment:
This was kind of evil, we were creating a new string for all of the content here even when we didn't need to have the full content any more.